### PR TITLE
PHR-13263 - Fix popup window

### PR DESCRIPTION
**Pull request by BalintViragJ**
[PHR-13263](https://pkbdev.atlassian.net/browse/PHR-13263) Fix popup window

[PHR-13263]: https://pkbdev.atlassian.net/browse/PHR-13263?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by dorogimate** [comment]
Manual testing: little bit different implementation from the ticket description, but seems to be working correctly. Edit window appears in the middle of the screen and cannot open multiple edit windows at the same time.

Two notes:

- One thing that would be good if the patient can only interact with the edit window, so cannot scroll and click on other buttons in the editor like remove or move up
- Editor window stays there when switch to different tab - Generated or Preview

**Review by dorogimate — APPROVED** [review]
LGTM

**Comment by BalintViragJ** [comment]
@dorogimate 

> One thing that would be good if the patient can only interact with the edit window, so cannot scroll and click on other buttons in the editor like remove or move up

But in the ticket this is one of the ACs: The edit window is not too large, so that the user can easily access the other parts of the page. I think this is controversial

> Editor window stays there when switch to different tab - Generated or Preview


This one will be done in: https://pkbdev.atlassian.net/browse/PHR-13271

File: src/app/elements/elements.component.ts [hunk 1]
```diff
@@ -19,7 +19,8 @@ export class ElementsComponent implements OnInit {
 
   public components = new Map<string, ComponentRef<any>>();
   private indexNumber = 0;
-  public compts: any[] = []
+  public isDialogOpen: boolean = false;
+
   private componentTypes: any = {
```
Comment (idasbiste) [inline src/app/elements/elements.component.ts:24]: Suggestion: create a Map that contains the component types already associated to the corresponding component. Then in `getComponentType` you just use the get function of Map.
```typescript
private componentTypes: Map<string, Type<any>> = new Map([
    ['textElementComponent', TextElementComponent], ...
]);
```

**Pull request merged**